test/e2e/upgrade_test.go

```diff
@@ -151,33 +206,3 @@ func (tc *testContext) scaleWMCODeployment(desiredReplicas int32) error {
 
 	return err
 }
-
-// deployWindowsWorkloadAndTester tests if the Windows Webserver deployment is available.
-// This is achieved by creating a Job object that continuously curls the webserver every 5 seconds.
-func (tc *testContext) deployWindowsWorkloadAndTester() (*batchv1.Job, error) {
```

Comment:
Please call out in the commit message that the defers were causing an issue, so you need to break this logic out of this function.

Reply:
This could have still remained a helper method that took affinity as an input. You could have then introduced another method that cleans up the resources created. We should always think more deeply before removing modularization and inlining a chunk of code.

Reply:
That's a fair point, +1 for modularization, but this is an outlier; 

Affinity is no longer needed, so it's out of the equation. Will remove the complex "affinity business" in follow-up PR;

The chunk of code inlined here is concise, and with the modularization of the [LinuxWorkloadTesterCommand](https://github.com/openshift/windows-machine-config-operator/pull/645/commits/c85df6109de0af3485ae3583b98b27c207f06f19#diff-fc8ea8c7c1a732fd9a425771cb3463d1cd143e5527acec751f32ba88f2df697aR57) generation. it aligns better in the parent execution thread, rather than passing in-context parameters back in order to defer the cleanup of the created resources.

Reply:
>The chunk of code inlined here is concise

Well your definition of concise differs from mine 😄

Reply:
>Affinity is no longer needed, so it's out of the equation. Will remove the complex "affinity business" in follow-up PR;

And given you are doing this, please reintroduce the function. It is always easier to read:
```go
	for x := range y {
		foobarstuff()
	}
```
rather than
```go
	for x := range y {
		// foo
		// bar
		// stuff
	}
```
Doesn't matter your definition of concise 😉

Reply:
Agree, and again +1 for modularization. You may be missing the [main reason for the refactoring](https://github.com/openshift/windows-machine-config-operator/pull/645#discussion_r700542776): bring the content of `deployWindowsWorkloadAndTester()` to the parent thread given the content of the function is simple, to ensure the deletion of the workloads happen after the node upgrade is complete and tested.

In layman's terms, reads better

```go
		foo := Foo()
                defer close(foo)
		bar := Bar()
                defer close(bar)
```

rather than

```go
		foo, bar := FooBar()
                defer close(foo)
                defer close(bar)
```

where `foo` and `bar` are orthogonal.

Reply:
OK, I understand now as to why you removed the function. However making the argument that I want use defer, so I will not moduralize is going to put us in a place where we will have ever growing functions. This seems to be case where the test needs to have a setup and teardown methods. Given the entities being created are finite, we don't need `foo, bar := FooBar()` but rather `createFooBar()` and `deleteFooBar()` given we know what is being created and what needs to be destroyed now.

Reply:
However this fix does not explain how the changed code will expose errors like we were seeing with the bug in place. Don't we need to check after the upgrade is complete that the workload continues to run and function?

Reply:
Yes, the logic is there already, just need a minor [adjustment](https://github.com/openshift/windows-machine-config-operator/pull/645#discussion_r701947458).

Reply:
>Given the entities being created are finite, we don't need foo, bar := FooBar() but rather createFooBar() and deleteFooBar() given we know what is being created and what needs to be destroyed now.

Correct, entities are finite and created with randomly generated names. The later, may jeopardize the happy flow, since we need to keep track of the generated names, preferred over increasing the numbers of  API calls to fetch the resources.

Reply:
I will let this go for now but I don't want to see this pattern of demodularizing repeated in the future.

Reply:
Agree, that's an antipattern.